modules/state_machine/states/walk.gd

```diff
@@ -0,0 +1,17 @@
+extends 'state.gd'
+
+func handle_input(actor, event):
+	if event.is_action_released("right") and actor.direction == 1:
+		actor.stop()
+	elif event.is_action_released("left") and actor.direction == -1:
+		actor.stop()
+
```

Comment:
Add condition to action press, so the character is able to change directions when walking